lib/api/element-commands/isVisible.js

```diff
@@ -51,8 +51,9 @@ class IsVisible extends BaseElementCommand {
   protocolAction() {
     if (this.api.capabilities) {
       // FIXME: temporary fix to get safari working
-      const {browserName} = this.api.capabilities;
-      if (browserName.toLowerCase() === 'safari') {
+      const {browserName, browserVersion} = this.api.capabilities;
+      // Are we running Safaridriver locally and is it Safari 12 or newer?
+      if (browserName.toLowerCase() === 'safari' && this.settings.webdriver.start_process === true && parseInt(browserVersion, 10) >= 12) {
```

Comment:
Why do you think the start_process check is necessary?

Reply:
It was part of the suggestion in #2308. I can use a "real browser" on a device farm like Browserstack and it still triggers this issue. I've not tested with other device farms, but I'd expect a similar issue. I've also not tried to spin up a local grid using safaridriver yet, and I don't know how common those kinds of grid setups are.

I'm open to better suggestions. We could listen for a custom browser capability instead of using `start_process`, for instance. This might then be easier to communicate with 3rd party vendors like Browserstack who provide generated configuration for Nightwatch to add and remove the capability to browsers as necessary.

Reply:
I think it's fine to remove the start_process check if we check for the version, because any safari driver which works with version 12 or higher will have that functionality. The only issue remaining is that the `browserVersion` capability was called `version` in earlier versions, so the `parseInt(browserVersion, 10) >= 12` check above won't work.

Reply:
I've tested removing the `start_process` check, and the error still presents when requesting a Safari 12+ browser on Browserstack, keeping the check resolves that issue. This is why I'm wondering if a custom capability we can check against instead would be preferable.

As for `browserVersion`/`version`, I'm not sure if you're referring to earlier versions of nightwatch or not. Has this check been backported to earlier versions of nightwatch too?

Reply:
> As for browserVersion/version, I'm not sure if you're referring to earlier versions of nightwatch or not. Has this check been backported to earlier versions of nightwatch too?

No, I've just checked browserstack with a default safari and got back a response with a Safari 6 which doesn't have a browserVersion field. So in any case, we should check for both `version` and `browserVersion` fields.

Reply:
Ok, I'll take another look at this tomorrow, thanks!

Reply:
Ok, I'm still struggling with this. The problem I'm having with browserstack is using a version of Safari version 12 or above. As expected (despite browserstack calling this capability `"browser_version"` instead of of `"browserVersion"`) it's falling into the case, but the browser returns an error. This was why I added the `start_process` check in the first place, as it causes Browserstack instances to not enter the conditional that causes it to fail, and then the test run succeeds.

So that suggests something weird going on with how Browserstack sets up newer Safari instances, and I still haven't had the opportunity to compare against other providers like SauceLabs.

I'm still thinking adding a custom capability or something is the best option to just force a certain type of behaviour where necessary, but unsure how to pass this through to the api layer. Thoughts @beatfactor?